lms/views/api/vitalsource.py

```diff
@@ -1,32 +1,39 @@
+import re
+
+from pyramid.httpexceptions import HTTPClientError
 from pyramid.view import view_config, view_defaults
 
 from lms.security import Permissions
-from lms.services.exceptions import ProxyAPIError
-from lms.views.api.vitalsource_sample_data import book_data, toc_data
 
 
 @view_defaults(renderer="json", permission=Permissions.API)
 class VitalSourceAPIViews:
     def __init__(self, request):
         self.request = request
+        self.vitalsource_service = request.find_service(name="vitalsource")
 
     @view_config(route_name="vitalsource_api.books.info")
     def book_info(self):
         book_id = self.request.matchdict["book_id"]
+        if not self._is_valid_book_id(book_id):
+            raise HTTPClientError("Invalid `book_id`. It must only contain [0-9A-Z-].")
```

Comment:
It looks like this error is resulting in the caller receiving an HTML rather than JSON error response:

```html
<!doctype html>
<html lang=en>
  <head>
    ...
  </head>
  <body>
    <div class="content">
      
  <main class="modal-content u-horizontally-center-children">
    <img src="/static/images/sad-annotation.svg">
    <p>
      Invalid `book_id`. It must only contain [0-9A-Z-].
    </p>
  </main>

    </div>

    
      
      
    
  </body>
</html>
```

Reply:
My bad, switched to json.

Here's is how it looks on the UI if that were to happen (which would only occour if the validation on the backend and frontned diverged)
![Screenshot from 2021-07-30 11-46-34](https://user-images.githubusercontent.com/1433832/127636152-a549ab48-9fb2-44af-92ff-a3317e50b593.png)

I'd say it's an ok message if we for example got it in a bug report.